File: app/src/main/java/com/jonasgerdes/stoppelmap/usecases/map/search/SearchResultHolder.java
package com.jonasgerdes.stoppelmap.usecases.map.search;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.jonasgerdes.stoppelmap.model.map.search.SearchResult;

/**
 * Created by Jonas on 27.07.2016.
 */
public abstract class SearchResultHolder extends RecyclerView.ViewHolder {

    public SearchResultHolder(View itemView) {
        super(itemView);
    }

    public abstract void onBind(SearchResult result);
}

File: app/src/main/java/com/jonasgerdes/stoppelmap/usecase/map/view/search/SearchResultHolder.kt
package com.jonasgerdes.stoppelmap.usecase.map.view.search

import android.support.v7.widget.RecyclerView
import android.view.View
import com.jonasgerdes.stoppelmap.model.entity.map.search.MapSearchResult

/**
 * @author Jonas Gerdes <dev@jonasgerdes.com>
 * @since 06.07.17
 */
abstract class SearchResultHolder<in E : MapSearchResult>(itemView: View)
    : RecyclerView.ViewHolder(itemView) {

    abstract fun onBind(result: E)
}